src/v/kafka/server/handlers/topics/types.cc

```diff
@@ -109,10 +109,20 @@ get_string_value(const config_map_t& config, std::string_view key) {
 static std::optional<bool>
 get_bool_value(const config_map_t& config, std::string_view key) {
     if (auto it = config.find(key); it != config.end()) {
-        return string_switch<std::optional<bool>>(it->second)
-          .match("true", true)
-          .match("false", false)
-          .default_match(std::nullopt);
+        try {
+            // Ignore case.
+            auto str_value = it->second;
+            std::transform(
+              str_value.begin(),
+              str_value.end(),
+              str_value.begin(),
+              [](const auto& c) { return std::tolower(c); });
+            return string_switch<std::optional<bool>>(str_value)
+              .match("true", true)
+              .match("false", false);
+        } catch (const std::runtime_error&) {
+            throw boost::bad_lexical_cast();
```

Comment:
I was following the same pattern I was familiar with from https://github.com/redpanda-data/redpanda/blob/df1e2474b14a790d75157e448e0dc1a57ede29f8/src/v/kafka/server/handlers/configs/config_utils.h#L654-L656, but we can certainly implement it in the way you described as well if you feel strongly. What do you think?

Reply:
I don't think `string_switch` has a strict contract per se, but if https://github.com/redpanda-data/redpanda/blob/497c5de0ea5bdd7483ecc625be872fdfd5ba21c0/src/v/strings/string_switch.h#L195-L202 were to change, I think it would break a few existing spots in the code.